python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_relative.py

```diff
@@ -656,193 +564,46 @@ def _analyse_gromacs(work_dir=None, estimator=None):
         if estimator not in ['MBAR','TI']:
             raise ValueError("'estimator' must be either 'MBAR' or 'TI'.")
 
-        # Figure out the gromacs version.
-        def gromacs_version():
-            p = _subprocess.run([_gmx_exe,'-version'], stdout=_subprocess.PIPE).stdout.decode('utf-8')
-            with _tempfile.TemporaryDirectory() as tmpdirname:
-                file = f'{tmpdirname}/version.txt'
-                with open (file,'w') as f:
-                    print(p, file=f)
-                f = open(file,'r')
-                for line in f:
-                    if "GROMACS version" in line:
-                        l = line.strip().split(':')
-                        l = l[1].strip()
-                        l = float(l)
```

Comment:
I appreciate that this is likely not your edit, but can we not use a single character `l` as a variable name. Depending on the font, this will look identical to `1`. Here we could just safely re-use `line`. (Although it seems a bit weird for the same variable to refer to multiple different things, i.e. the full line, the split line as a list, the first record with no preceding/trailing whitespace, the first record as a float. Perhaps `tmp` would be best to identify that it's a temporary variable.

Reply:
Sorry, what do you mean here?